XdistroSetup.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+set -eu -o pipefail
+#
+# Script to automate basic preparation of a cross-distro
+# bootstrap-builder host for a given alternate-distro
+# build-target
+#
+#################################################################
+PROGNAME=$( basename "$0" )
+RUNDIR="$( dirname "$0" )"
+DEBUG="${DEBUG:-UNDEF}"
+
+# Make interactive-execution more-verbose unless explicitly told not to
+if [[ $( tty -s ) -eq 0 ]] && [[ ${DEBUG} == "UNDEF" ]]
+then
+   DEBUG="true"
+fi
+
+
+# Error handler function
+function err_exit {
+   local ERRSTR
+   local ISNUM
+   local SCRIPTEXIT
+
+   ERRSTR="${1}"
+   ISNUM='^[0-9]+$'
+   SCRIPTEXIT="${2:-1}"
+
+   if [[ ${DEBUG} == true ]]
+   then
+      # Our output channels
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+
+# Print out a basic usage message
+function UsageMsg {
+   local SCRIPTEXIT
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-d' 'Distro nickname (e.g., "Rocky")'
+      printf '\t%-4s%s\n' '-h' 'Print this message'
+      printf '\t%-4s%s\n' '-k' 'List of RPM-validation key-files or RPMs'
+      printf '\t%-4s%s\n' '-r' 'List of repository-related RPMs'
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--distro-name' 'See "-d" short-option'
+      printf '\t%-20s%s\n' '--help' 'See "-h" short-option'
+      printf '\t%-20s%s\n' '--repo-rpms' 'See "-r" short-option'
+      printf '\t%-20s%s\n' '--sign-keys' 'See "-k" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+# Install the alt-distro's GPG key(s)
+function InstallGpgKeys {
+   local LOOP
+   LOOP=0
+
+   while [[ LOOP -lt ${#PKGSIGNKEYS[*]} ]]
```

Comment:
It seems like a for loop would be a lot cleaner here

Reply:
While I've never actually run into RPMs with embedded spaces, I always assume the worst: array-iteration avoids the whole "you need to surround `${ARRAY[@]}` with quotes" error from `shellcheck`

Reply:
Right, so just `for ITEM in "${ARRAY[@]}"`? That expands each item in its own quoted string.